sw/device/tests/sim_dv/lc_ctrl_transition_test.c

```diff
@@ -21,6 +21,21 @@ static dif_lc_ctrl_t lc;
 
 const test_config_t kTestConfig;
 
+/**
+ * Track number of transactions under this test.
+ *
+ * From the software / compiler's perspective, this is a constant (hence the
+ * `const` qualifier). However, the external DV testbench finds this symbol's
+ * address and modifies it via backdoor, to track how many transactions have
+ * been sent. Hence, we add the `volatile` keyword to prevent the compiler from
+ * optimizing it out.
+ * The `const` is needed to put it in the .rodata section, otherwise it gets
+ * placed in .data section in the main SRAM. We cannot backdoor write anything
+ * in SRAM at the start of the test because the CRT init code wipes it to 0s.
+ */
+static volatile const uint8_t kTestIdx = 0x0;
```

Comment:
nice. does this mean we will a lot of the handshaking between DV / C will fall into this cateogry?

Reply:
Thanks, i copied this from Weicai's UART test! I think Sri implemented this to pass array or data from SV to C testbench:)

Reply:
Yes, this is ONE way to pass info from the testbench to SW, without the SW (or the compiler) even knowing about it, but it is a DV specific methodogy. It helps us completely randomize the SW test and thus get more coverage.

For usecases where there is actual transfer of data between the host (OTTool / UVM TB) using actual peripherals such as UART/GPIOs/I2C. @timothytrippel is working on standardizing that interface across on-device OTTF and on-host OTTool/UVM TB such that even the test involving IO peripherals can be shared across simulation platforms, allowing us to develop truly universal tests and share the testplan in its entirety across DV, FPGA and post-Si.